src/mokernels/slfm.jl

```diff
@@ -0,0 +1,67 @@
+@doc raw"""
+    LatentFactorMOKernel(
+        g::AbstractVector{<:Kernel}, 
+        e::AbstractVector{<:Kernel}, 
+        A::AbstractMatrix
+        )
+
+A semiparametric kernel for problems involving multiple response variables.
+
+``k((x, p), (y, p)) = k_p(x, y) = \Sum^{Q}_{q=1} A_{pq}g_q(x, y) + e_p(x, y)``
+
+# Arguments
+- `g::AbstractVector{<:Kernel}`: an array of kernels
+- `e::AbstractVector{<:Kernel}`: an array of kernels
+- `A::AbstractMatrix`: an matrix of weights for the kernels of shape (length(e), length(g))
+
+
+# Reference:
+- [Seeger, Teh, and Jordan (2005)](https://infoscience.epfl.ch/record/161465/files/slfm-long.pdf)
+
+"""
+struct LatentFactorMOKernel{
+    Tg <:AbstractVector{<:Kernel}, 
+    Te <:AbstractVector{<:Kernel}, 
+    TA <: AbstractMatrix
+    } <: Kernel
+    g::Tg
+    e::Te
+    A::TA
+    function LatentFactorMOKernel(
+        g::AbstractVector{<:Kernel}, 
+        e::AbstractVector{<:Kernel}, 
+        A::AbstractMatrix
+        )
+        @assert (length(e), length(g)) == 
+            size(A) "Size of A not compatible to the given array of kernels"
```

Comment:
IIRC `@assert` shouldn't be used for checks that are critical since it can potentially be removed by the compiler.
```suggestion
        (length(e), length(g)) == size(A)
            error("size of `A` is not compatible with the provided kernels")
```